Automattic/jetpack
# AI Assistant: Support upgrade prompt for Simple and Atomic sites

Part of https://github.com/orgs/Automattic/projects/667/views/1?pane=issue&itemId=32070782

Depends on D114991-code. Upgrade prompts will only be visible after D114991-code deployment.

## Proposed changes:

* Uses the `useUpgradeFlow` hook for non-Jetpack websites
* Removes the condition on being a Jetpack site for the `$require_upgrade` flag

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* On a JN site, send 20 requests for the AI Assistant
* Check that the upgrade flow works and the user is initially redirected to a price page
* On a Simple or Atomic site, check that the upgrade prompt is displayed after 20 requests and the user is redirected directly to the checkout page
